reVX/plexos/plexos_plants.py

```diff
@@ -0,0 +1,1006 @@
+        plants_meta.index.name = 'plant_id'
+
+        plants_meta = self.plexos_table.merge(plants_meta.reset_index(),
+                                              on='plant_id', how='outer')
+
+        if out_fpath:
+            plants_meta.to_csv(out_fpath, index=False)
+
+        return plants_meta
+
+    def _make_plant_meta(self, bus_meta):
+        """
+        Create plant meta data for given bus
+
+        Parameters
+        ----------
+        bus_meta : pandas.Series
+            Meta data for desired bus to build plant for
+
+        Returns
+        -------
+        plant_meta : pandas.DataFrame
+            Meta data for plant associated with given bus, constructed from:
+            - Plant table
+            - Supply Curve table
+            - Bus capacity
+        """
+        plant_meta = pd.concat(self[bus_meta['plant_id']], axis=1).T
+        plant_meta['gen_gids'] = \
+            plant_meta['res_gids'].apply(lambda gids: [self.cf_gid_map[gid]
+                                                       for gid in gids])
+
+        sc_cols = ['res_gids', 'gen_gids', 'gid_counts', 'capacity']
+        sc_cols = [c for c in self.sc_table if c not in sc_cols]
+        plant_meta = plant_meta.merge(self.sc_table[sc_cols],
+                                      on='sc_gid', how='left')
+
+        plant_capacity = plant_meta['build_capacity'].sum()
+        if plant_capacity != bus_meta['capacity']:
+            bulid_capacity = (plant_meta['build_capacity'] / plant_capacity
+                              * bus_meta['capacity'])
+            plant_meta.loc[:, 'build_capacity'] = bulid_capacity
+
+        return plant_meta
+
+    @staticmethod
+    def _make_profile(cf_fpath, plant_build):
+        """
+        Make generation profiles for given plant buildout
+
+        Parameters
+        ----------
+        cf_fpath : str
+            Path to reV Generation output .h5 file to pull CF profiles from
+        plant_build : pandas.DataFrame
+            DataFrame describing plant buildout
+
+        Returns
+        -------
+        profile: ndarray
+            Generation profile for plant as a vector
+        """
+        with Resource(cf_fpath) as f:
+            profile = None
+            for _, row in plant_build.iterrows():
+                gid_capacities = (row['gid_counts'] / np.sum(row['gid_counts'])
+                                  * row['build_capacity'])
+                cf_profiles = f['cf_profile', :, row['gen_gids']]
+                for i, cf_profile in enumerate(cf_profiles.T):
+                    if profile is None:
+                        profile = cf_profile * gid_capacities[i]
+                    else:
+                        profile += cf_profile * gid_capacities[i]
+
+        if len(profile.shape) != 1:
+            profile = profile.flatten()
+
+        return profile
+
+    def aggregate_plants(self, out_fpath):
+        """
+        Aggregate plants from capacity factor profiles and save to given
```

Comment:
You're really aggregating the profiles right? Making aggregated power profiles from the source resource profiles once the "allocation" step is done? The naming and descriptions are confusing. In a vacuum, I would have no idea when to use this method.